### PR TITLE
Reimplement more of the closure var rename step

### PR DESCRIPTION
**Pull request by u9g**
https://github.com/google/closure-compiler/blob/c383a3a1d2fce33b6c778ef76b5a626e07abca41/src/com/google/javascript/jscomp/RenameVars.java#L475C10-L483

```diff
  Original   -> Minified   -> Gzip      
- 10.82 MB   -> 3.52 MB    -> 909.47 kB typescript.js
+ 10.82 MB   -> 3.52 MB    -> 902.66 kB typescript.js
```

### CODE REVIEW

File: crates/oxc_minifier/src/mangler/mod.rs [hunk 1]
```diff
@@ -170,20 +170,41 @@ impl ManglerBuilder {
...
+            });
+        }
+
+        let mut freq_iter = frequencies.iter();
+
+        for slice_of_same_len_strings in names.group_by_mut(|a, b| a.len() == b.len()) {
+            let mut symbols_renamed_in_this_batch =
+                freq_iter.by_ref().take(slice_of_same_len_strings.len()).collect::<Vec<_>>();
+
+            debug_assert!(symbols_renamed_in_this_batch.len() == slice_of_same_len_strings.len());
```
Comment (u9g) [inline crates/oxc_minifier/src/mangler/mod.rs:193]: not strictly necessary, could remove

Comment (Boshen) [reply crates/oxc_minifier/src/mangler/mod.rs:193]: This is good, don't remove it.

**Review by Boshen — COMMENTED** [review]
It would be great if there are some line by line explanations to expand on what the comment from google closure compiler is talking about.

Can you also reference the link in the comments [google/closure-compiler@c383a3a/src/com/google/javascript/jscomp/RenameVars.java#L475C10-L483](https://github.com/google/closure-compiler/blob/c383a3a1d2fce33b6c778ef76b5a626e07abca41/src/com/google/javascript/jscomp/RenameVars.java#L475C10-L483)

We want the next person to gain maximal context when reading these kind of code.

**Comment by Boshen** [comment]
Need to fix clippy as well 😀

**Review by Boshen — APPROVED** [review]

**Pull request merged**